p4-16/spec/scripts/trim-p4-grammar-file.py

```diff
@@ -0,0 +1,136 @@
+#! /usr/bin/env python3
```

Comment:
the robust way is to use a real parser to get a representation of the grammar.
the bnf language grammar isn't that complicated.
I have used the parglare parser generator which is written and generates Python, it's relatively easy to use.
http://www.igordejanovic.net/parglare/0.16.0/

Reply:
Anyone else is welcome to create a robust solution if they want.  This code is tested and effective with the current grammar.

Reply:
Do you think it is reasonable to merge this in as is without support for general Bison grammars?  These programs work with the current grammar, and any likely changes I ever expect to see in the P4 grammar files, but I can look into fixing them if changes are made to these grammars in the future that break these programs (that seems unlikely to me, though).